components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/ApplicationAuthenticatorService.java

```diff
@@ -67,10 +67,41 @@ public static ApplicationAuthenticatorService getInstance() {
         return instance;
     }
 
+    /**
+     * This method is used to get the list of SYSTEM defined local authenticator configurations.
+     *
+     * @deprecated It is recommended to use {@link #getAllSystemDefinedLocalAuthenticators()},
+     * which returning the SYSTEM defined local application authenticator configurations.
+     */
+    @Deprecated
     public List<LocalAuthenticatorConfig> getLocalAuthenticators() {
         return this.localAuthenticators;
     }
 
+    /**
+     * This returns list of all SYSTEM defined local authenticator configurations.
+     *
+     * @return Retrieved LocalAuthenticatorConfig.
+     */
+    public List<LocalAuthenticatorConfig> getAllSystemDefinedLocalAuthenticators() {
+
+        return this.localAuthenticators;
+    }
+
+    /**
+     * This returns list of all SYSTEM and USER defined local authenticator configurations.
+     *
+     * @param tenantDomain  Tenant domain.
+     * @return Retrieved LocalAuthenticatorConfig.
+     */
+    public List<LocalAuthenticatorConfig> getAllLocalAuthenticators(String tenantDomain)
+            throws AuthenticatorMgtException {
+
+        List<LocalAuthenticatorConfig> configList = new ArrayList<>(getAllUserDefinedLocalAuthenticators(tenantDomain));
+        configList.addAll(localAuthenticators);
```

Comment:
At a given point of time looks like this won't return all system defined authenticators. This is because system defined authenticators are provided an input to the service.
This is fine for the moment as that is how this service is defined

Reply:
The system defined local authenticators are added to localAuthenticators list at the server start up, and that list will not changes after that.